src/Sylius/Bundle/AdminBundle/Controller/DashboardController.php

```diff
@@ -71,9 +72,15 @@ public function getRawData(Request $request): Response
     private function findChannelByCodeOrFindFirst(?string $channelCode): ?ChannelInterface
     {
         if (null !== $channelCode) {
-            return $this->channelRepository->findOneByCode($channelCode);
+            $channel = $this->channelRepository->findOneByCode($channelCode);
+            Assert::nullOrIsInstanceOf($channel, ChannelInterface::class);
```

Comment:
I'm just wondering if there are any pros to having assertions in such cases. We are fairly sure it is a `ChannelInterface` because we use a channel repository. So maybe the `@var ChannelInterface|null` annotation is enough. Thanks to this we get rid of some extra if statements. WDYT?

Reply:
> https://github.com/phpstan/phpstan-doctrine
> Interprets `EntityRepository<MyEntity>` correctly in phpDocs for further type inference of methods called on the repository.
> Recognizes magic findBy*, findOneBy* and countBy* methods on EntityRepository.

Do you guys think it does what I think it does? (Didn't test) That would remove need to define magic methods for typehint, like so
https://github.com/Sylius/Sylius/blob/2818ace6223948f9b91a95aa3a41c0b6bea0d718/src/Sylius/Bundle/ChannelBundle/Doctrine/ORM/ChannelRepository.php#L32-L35

and if we're to do a `phpstan-sylius-resource` extension, which will read current channel class from resource configuration (They do read doctrine mappings as well https://github.com/phpstan/phpstan-doctrine/tree/1.3.x/src/Doctrine/Mapping) and supply actual class to this `EntityRepository<MyEntity>` value, then we shouldn't need to add asserts or @var each time repository is used.

Reply:
Sorry, I didn't answer this question... Of course, there is some misconfiguration in the type documented in the methods or in the way PHPStan, Psalm, and/or PHPStorm recognize these objects, but it is not the purpose of this PR. Probably another issue should be opened and this discussion should be moved there.